pygmt/src/triangulate.py

```diff
@@ -0,0 +1,382 @@
+        b="binary",
+        d="nodata",
+        e="find",
+        f="coltypes",
+        h="header",
+        i="incols",
+        r="registration",
+        s="skiprows",
+        w="wrap",
+    )
+    @kwargs_to_strings(I="sequence", R="sequence", i="sequence_comma")
+    def _triangulate(
+        data=None, x=None, y=None, z=None, output_type=None, outfile=None, **kwargs
+    ):
+        """
+        Delaunay triangulation or Voronoi partitioning and gridding of
+        Cartesian data.
+
+        Must provide ``outfile`` or ``outgrid``.
+
+        Full option list at :gmt-docs:`triangulate.html`
+
+        {aliases}
+
+        Parameters
+        ----------
+        x/y/z : np.ndarray
+            Arrays of x and y coordinates and values z of the data points.
+        data : str or {table-like}
+            Pass in (x, y, z) or (longitude, latitude, elevation) values by
+            providing a file name to an ASCII data table, a 2D
+            {table-classes}.
+        {J}
+        {R}
+        {I}
+        outgrid : bool or str
+            The name of the output netCDF file with extension .nc to store the
+            grid in. The interpolation is performed in the original
+            coordinates, so if your triangles are close to the poles you are
+            better off projecting all data to a local coordinate system before
+            using ``triangulate`` (this is true of all gridding routines) or
+            instead select :gmt-docs:`sphtriangulate <sphtriangulate.html>`.
+        outfile : str or bool or None
+            The name of the output ASCII file to store the results of the
+            histogram equalization in.
+        output_type: str
+            Determines the output type. Use "file", "xarray", "pandas", or
+            "numpy".
+        {V}
+        {b}
+        {d}
+        {e}
+        {f}
+        {h}
+        {i}
+        {r}
+            Only valid with ``outgrid``.
+        {s}
+        {w}
+
+        Returns
+        -------
+        ret: numpy.ndarray or pandas.DataFrame or xarray.DataArray or None
+            Return type depends on the ``output_type`` parameter:
+
+            - numpy.ndarray if ``output_type`` is "numpy"
+            - pandas.DataFrame if ``output_type`` is "pandas"
+            - xarray.DataArray if ``output_type`` is "xarray""
+            - None if ``output_type`` is "file" (output is stored in
+              ``outgrid`` or ``outfile``)
+        """
+        with Session() as lib:
+            # Choose how data will be passed into the module
+            table_context = lib.virtualfile_from_data(
+                check_kind="vector", data=data, x=x, y=y, z=z, required_z=False
+            )
+            with table_context as infile:
+                if "G" not in kwargs:  # table output if outgrid is unset
+                    kwargs.update({">": outfile})
+                else:  # NetCDF or xarray.DataArray output if outgrid is set
+                    outgrid = kwargs["G"]
```

Comment:
```suggestion
                if (outgrid := kwargs.get("G")) is None:  # table output if outgrid is unset
                    kwargs.update({">": outfile})
```
I think we should use `kwargs.get(arg)` rather than `if arg not in kwargs` to avoid the ongoing issue when an argument is explicitly set as ``None``.

Reply:
Oo, the Python 3.8+ walrus operator! Very fancy!